### ISSUE TITLE
Strange bug with options --tiny 

### ISSUE THREAD

**Issue by FAMILIAR-project** [TuxML/ProjetIrma#119]
I've tried:
`python3 kernel_generator.py --tiny --dev 1`

and the trace is puzzling: 
```
00:00:21 [2018-11-16 13:54:26 CET] [gen_config] Randomising new KCONFIG_FILE
00:00:21 [2018-11-16 13:54:26 CET] [gen_config] "tiny_tuxml.config" used
00:00:21 [2018-11-16 13:54:26 CET] [gen_config] KCONFIG_ALLCONFIG=/TuxML/tiny_tuxml.config make -C /TuxML/linux-4.13.3 randconfig
00:00:23 KCONFIG_SEED=0xCAB59792
```

It seems it does not enter here:
https://github.com/TuxML/ProjetIrma/blob/dev/core/tuxml.py#L219-L227

the condition test seems correct though: 
`if tset.TUXCONFIG == "tiny_tuxml.config":`

I don't understand. 
I've checked that the Docker image contains the good code (branch dev), it's clearly the case. 

I'm missing something here

**Comment by FAMILIAR-project** [comment]
Investigation so far:
 * I've tried to call `tuxml.py` individually (within the Docker), it correctly handles the tiny case 
 * I've tried to call `runandlog` (within the Docker), success
 * I've tried to exec `runandlog` outside docker with something like `docker exec -t c9a8a29dd3d5 /TuxML/runandlog.py 0 --tiny` almost a success (see below: tinyconfig is used but there is an error at the end of the compilation)


```
00:00:09 [2018-11-16 16:52:19 CET] [gen_config] Randomising new KCONFIG_FILE
00:00:09 [2018-11-16 16:52:19 CET] [gen_config] "tiny_tuxml.config" used
00:00:09 [2018-11-16 16:52:19 CET] [gen_config] TINY config with preset values here=/TuxML/x64 make -C /TuxML/linux-4.13.3 tinyconfig
00:00:13 [2018-11-16 16:52:23 CET] [compilation] Compilation in progress
00:00:43 [2018-11-16 16:52:53 CET] [compilation] Compilation done
00:00:43 [2018-11-16 16:52:53 CET] [launcher] Successfully compiled in 00:00:30
00:00:43 [2018-11-16 16:52:53 CET] [boot_try] Launching boot test on kernel
00:00:43 WARNING: missing /lib/modules/4.9.93-linuxkit-aufs
00:00:43 Ensure all necessary drivers are built into the linux image!
00:00:43 depmod: ERROR: could not open directory /lib/modules/4.9.93-linuxkit-aufs: No such file or directory
00:00:43 depmod: FATAL: could not search modules: No such file or directory
00:00:43 depmod: WARNING: could not open /var/tmp/mkinitramfs_SoF10E/lib/modules/4.9.93-linuxkit-aufs/modules.order: No such file or directory
00:00:43 depmod: WARNING: could not open /var/tmp/mkinitramfs_SoF10E/lib/modules/4.9.93-linuxkit-aufs/modules.builtin: No such file or directory
00:00:44 Could not initialize SDL(No available video device) - exiting
00:02:39 [2018-11-16 16:54:49 CET] [boot_try] More than 60 attempts at reading, possible infinite loop in boot process, interrupting
00:02:39 [2018-11-16 16:54:49 CET] [send_data] Sending compilation and tests results to database
00:02:39 [2018-11-16 16:54:49 CET] [get_compressed_sizes] Computing compressed kernels
00:02:39 [Errno 2] No such file or directory: '/TuxML/linux-4.13.3.config'
```

**Comment by FAMILIAR-project** [comment]
I think I get the problem: tuxml.py code is from tuxml/tuxmldebian instead of tuxml/debiantuxml

Problem is that the content of _IMAGE (uncompressed image) already contains tuxml.py or the Linux source folder. It seems NOT override when uncompressing the _COMPRESSED_IMAGE
https://github.com/TuxML/ProjetIrma/blob/dev/kernel_generator.py#L244-L259  

I don't understand why 
`tuxml/tuxmldebian` still contains the code and the source code

I think we should also rename the names of the images it's very confusing

**Comment by mipicard** [comment]
I agree with the renaming.
In fact, both debiantuxml and tuxmldebian contain the same code, but debiantuxml contains it within a tar while tuxmldebian contains it uncompressed.

I assure you that the image tuxmldebian is replaced if any change have been made, because it will uncompress debiantuxml each time you run the script WITHOUT --local argument (and for our case with --dev).

**Comment by FAMILIAR-project** [comment]
My point is that when uncompressing debiantuxml into tuxmldebian, it does not override existing scripts such as tuxml.py. I really have the impression we are actually running an old version of tuxml.py in the Docker image (and not the new one).

**Comment by FAMILIAR-project** [comment]
Within Docker debiantuxml, I've tried to decompress manually 

```
tar xf /TuxML/linux-4.13.3.tar.xz -C /TuxML 
tar xf /TuxML/TuxML.tar.xz -C /TuxML 
apt-get install -qq -y --no-install-recommends $(cat /dependencies.txt)
```

and then running `runandlog` with or without tiny.

With tiny, good news: it enters in the conditional statement about tiny. 
The bad news is that **with or without** tiny, I've this error at the end of the compilation: 
```
00:08:26 [Errno 2] No such file or directory: '/TuxML/linux-4.13.3.config'

Processing output.log ...
CID array: []
Cid unfound, no output.log file has been sent
```

With kernel generator and thus tuxmldebian, without tiny works well. 
My feeling is that we have the good code in debiantuxml, BUT not when building the full image and uncompressing it

**Comment by FAMILIAR-project** [comment]
I've eventually succeeded to enter in the tiny_tuxml condition! 
I've removed *all* Docker images (debiantuxml:dev, tuxmldebian:dev, debian:latest)

strange note: I've removed "only" debiantuxml and tuxmldebian, and it did not work. 
I've indeed a debian:latest on my computer, and I also needed to remove it. 
I've impression the fact to have debian:latest puts into troubles tuxmldebian. Very strange, we have to check it in upcoming versions
